### PR TITLE
Improve diagnostics and add code fixes for top-level await

### LINKED ISSUE #36036: Top-level `await` error message is extremely unclear
```ts
// @target: es2017
// @module: esnext

await 100;
```

This gives the following error message:

```
'await' outside of an async function is only allowed at the top level of a module when '--module' is 'esnext' or 'system' and '--target' is 'es2017' or higher.
```

This is really unclear! My compiler options are all set correctly, so why is this not working?

It's because it's not a module. This error message is expecting me to meet 3 different conditions, "top level of a module" is unclear for any user who just wants to use this feature. They'll likely take "module" to just mean "file".

### LINKED ISSUE #36171: "Implement interface" codefix missing newline after first member
**TypeScript Version:**  3.7.3

**Search Terms:** implement interface codefix

**Code**

The "Implement interface" codefix fails to add a newline character after the first newly-added member when the class body is split across multiple lines:

image

**Actual behavior:**

image

### PR DESCRIPTION
**Pull request by rbuckton**
This improves the diagnostic messages for top-level `await` and adds several code-fixes to improve the experience.

In addition, this fixes a small bug in our `textChanges` algorithm to ensure we correctly add new-lines when inserting nodes at the top of a class body/object literal (and elide trailing commas in an object literal when they aren't needed.

Fixes #36036
Fixes #36171

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Looks like there's a merge conflict.

___

@typescript-bot pack this

**Comment by DanielRosenwasser** [comment]
The message improvements look good.

I would suggest always keeping the trailing comma to be honest - but I happen to be especially prone to adding new fields after members that lack a comma..

**Comment by rbuckton** [comment]
@DanielRosenwasser: Trailing comma is illegal in JSON, so if we ever have a codefix that affects `package.json`, it would make that file invalid. I'd rather trim the trailing comma to be safe.

File: src/compiler/core.ts [hunk 1]
```diff
@@ -1369,6 +1374,7 @@ namespace ts {
         return typeof x === "number";
     }
 
+    // export function tryCast<TOut extends Exclude<TIn, undefined>, TIn = any>(value: TIn | undefined, test: (value: Exclude<TIn, undefined>) => value is TOut): TOut | undefined;
```
Comment (andrewbranch) [inline src/compiler/core.ts]: Meant to delete this?

**Comment by DanielRosenwasser** [comment]
Would it be too much trouble to have the codefix check if it's in ES3 or working under JSON?

**Comment by DanielRosenwasser** [comment]
or have the printer "do the right thing" under either mode?

**Comment by rbuckton** [comment]
I've updated it so that it only elides the trailing comma when the source file is a .json file (since there's no way to tell whether it's a normal .json or a non-standard .json like our tsconfig files).

**Pull request merged**